lib/superset/dashboard/export.rb

```diff
@@ -45,35 +72,58 @@ def params
 
       def save_exported_zip_file
         File.open(zip_file_name, 'wb') { |fp| fp.write(response.body) }
+        puts "Saved zip file: #{zip_file_name}"
       end
 
       def unzip_files
         @extracted_files = unzip_file(zip_file_name, tmp_uniq_dashboard_path)
+        puts "Unzipped files: #{@extracted_files.inspect}"
       end
 
       def download_folder
         File.dirname(extracted_files[0])
       end
 
+      def destination_path_with_dash_id
+        @destination_path_with_dash_id ||= File.join(destination_path, dashboard_id.to_s)
+      end
+
+      def clean_destination_directory
+        if Dir.exist?(destination_path_with_dash_id)
+          FileUtils.rm_rf(Dir.glob("#{destination_path_with_dash_id}/*"))
+          puts "Deleted all existing files in #{destination_path_with_dash_id}."
+        else
+          FileUtils.mkdir_p(destination_path_with_dash_id)
+          puts "Created new destination directory: #{destination_path_with_dash_id}"
+        end
+      end
+
       def copy_export_files_to_destination_path
-        path_with_dash_id = File.join(destination_path, dashboard_id.to_s)
-        FileUtils.mkdir_p(path_with_dash_id) unless File.directory?(path_with_dash_id)
+        Dir.glob("#{download_folder}/**/*").each do |item|
+          next if File.directory?(item) # Skip directories
+
+          relative_item_path = Pathname.new(item).relative_path_from(Pathname.new(download_folder)).to_s
+          destination_item = File.join(destination_path_with_dash_id, relative_item_path)
 
-        Dir.glob("#{download_folder}/*").each do |item|
-          FileUtils.cp_r(item, path_with_dash_id)
+          FileUtils.mkdir_p(File.dirname(destination_item)) unless Dir.exist?(File.dirname(destination_item))
+          puts "Copying #{item} to #{destination_item}"
+          FileUtils.cp(item, destination_item)
```

Comment:
best to revert this to the original cp_r 
as its cleaner

Reply:
Reverted: https://github.com/rdytech/superset-client/pull/40/commits/6e45818648453c83347db7ac5ea5b03076b5d15f